### PR TITLE
search: increase maxrecursiondepth for resolving query string

### PR DESCRIPTION
**Pull request by pazembrz**
* INSPIR-3377

### CODE REVIEW

File: backend/inspirehep/search/factories/query.py [hunk 1]
```diff
@@ -4,16 +4,27 @@
...
 
     def inspire_query(query_string, search):
-        return Q(inspire_query_parser.parse_query(query_string))
+        original_recursion_limit = sys.getrecursionlimit()
+        sys.setrecursionlimit(
+            current_app.config.get("SEARCH_MAX_RECURSION_LIMIT", 5000)
+        )
+        try:
+            query = Q(inspire_query_parser.parse_query(query_string))
+        except:
```
Comment (michamos) [inline backend/inspirehep/search/factories/query.py:24]: the `except` is useless here.

Comment (pazembrz) [reply backend/inspirehep/search/factories/query.py]: :+1: I forgot to remove it ;)

**Comment by pazembrz** [comment]
Decorator? I'm not fan of decorators...  Maybe as context manager instead?

**Comment by michamos** [comment]
context manager would be good :+1:

**Pull request merged**